presto-product-tests/src/main/java/com/facebook/presto/tests/iceberg/TestIcebergHiveMetadataListing.java

```diff
@@ -0,0 +1,92 @@
+public class TestIcebergHiveMetadataListing
+        extends ProductTest
+{
+    private List<QueryAssert.Row> preexistingTables;
+    private List<QueryAssert.Row> preexistingColumns;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        cleanUp();
+        preexistingTables = onPresto().executeQuery("SHOW TABLES FROM iceberg.default").rows().stream()
+                        .map(list -> row(list.toArray()))
+                                .collect(Collectors.toList());
+        preexistingColumns = onPresto().executeQuery("SELECT table_name, column_name FROM iceberg.information_schema.columns " +
+        "WHERE table_catalog = 'iceberg' AND table_schema = 'default'").rows().stream()
+                        .map(list -> row(list.toArray()))
+                                .collect(Collectors.toList());
+        onPresto().executeQuery("CREATE TABLE iceberg.default.iceberg_table1 (_string VARCHAR, _integer INTEGER)");
+        onPresto().executeQuery("CREATE TABLE hive.default.hive_table (_double DOUBLE)");
+        onPresto().executeQuery("CREATE VIEW hive.default.hive_view AS SELECT * FROM hive.default.hive_table");
+        onPresto().executeQuery("CREATE VIEW iceberg.default.iceberg_view AS SELECT * FROM iceberg.default.iceberg_table1");
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+    {
+        onPresto().executeQuery("DROP VIEW IF EXISTS iceberg.default.iceberg_view");
+        onPresto().executeQuery("DROP TABLE IF EXISTS iceberg.default.iceberg_table1");
+        onPresto().executeQuery("DROP VIEW IF EXISTS hive.default.hive_view");
+        onPresto().executeQuery("DROP TABLE IF EXISTS hive.default.hive_table");
+    }
+
+    @Test(groups = {ICEBERG, STORAGE_FORMATS})
+    public void testTableListing()
+    {
+        assertThat(onPresto().executeQuery("SHOW TABLES FROM iceberg.default"))
+                .containsOnly(ImmutableList.<QueryAssert.Row>builder()
+                        .addAll(preexistingTables)
+                        .add(row("iceberg_table1"))
+                        .add(row("iceberg_view"))
+                        .add(row("hive_view"))
+                        .add(row("hive_table"))
+                        .build());
+    }
+
+    @Test(groups = {ICEBERG, STORAGE_FORMATS})
+    public void testColumnListing()
+    {
+        assertThat(onPresto().executeQuery(
+                "SELECT table_name, column_name FROM iceberg.information_schema.columns " +
+                        "WHERE table_catalog = 'iceberg' AND table_schema = 'default'"))
+                .containsOnly(ImmutableList.<QueryAssert.Row>builder()
+                        .addAll(preexistingColumns)
+                        .add(row("iceberg_table1", "_string"))
+                        .add(row("iceberg_table1", "_integer"))
+                        .add(row("iceberg_view", "_string"))
+                        .add(row("iceberg_view", "_integer"))
+                        .add(row("hive_view", "_double"))
```

Comment:
@imjalpreet Thank you for the valuable suggestions, product tests are passing finally. I need to verify if we want hive_view to be listed as part of the above query? I feel this is not the desired behaviour. 
Trino takes care of this already - https://github.com/trinodb/trino/pull/8153/files#diff-d45108e44a12da55f3fbda8c8cb9f3ea323ae3caa7f4dddd20fa2535b499a31e